spring-data-jpa/src/main/java/org/springframework/data/jpa/mapping/JpaMetamodelMappingContext.java

```diff
@@ -171,25 +171,24 @@ public boolean isMetamodelManagedType(Class<?> type) {
 		 */
 		@Nullable
 		private Metamodel getMetamodelFor(Class<?> type) {
-
-			for (Metamodel model : metamodels) {
-
-				try {
+			Metamodel currentModel = null;
+			try {
+				for (Metamodel model : metamodels) {
+					currentModel = model;
 					model.managedType(type);
 					return model;
-				} catch (IllegalArgumentException o_O) {
-
-					// Fall back to inspect *all* managed types manually as Metamodel.managedType(…) only
-					// returns for entities, embeddables and managed superclasses.
-
-					for (ManagedType<?> managedType : model.getManagedTypes()) {
-						if (type.equals(managedType.getJavaType())) {
-							return model;
-						}
+				}
+			} catch (IllegalArgumentException o_O) {
+				
+				// Fall back to inspect *all* managed types manually as Metamodel.managedType(…) only
+				// returns for entities, embeddables and managed superclasses.
+				
+				for (ManagedType<?> managedType : currentModel.getManagedTypes()) {
```

Comment:
Looks like we may run into a potential NPE here.

Reply:
Thanks! @christophstrobl 

I made a mistake in adding the null check. 😢 
Based on your feedback, I have now corrected the null check.

Reply:
Hi! @christophstrobl 

Upon reevaluation, I believe there shouldn’t be an NPE occurrence as you mentioned.

To meet the conditions for an NPE as mentioned in the feedback, ‘metamodels’ within the try block would need to be null. However, since the constructor already handles this check, I anticipate no NPE occurrence within the catch block.

Note: There might be aspects I’m overlooking due to my current skill level, but based on my understanding, this is how I perceive it!!!